pseudospectral/spectra/derivative_1D.py

```diff
@@ -1,70 +1,85 @@
 import numpy as np
 import scipy.fft
 
+
 class Derivative1D:
     """
-    Class to represent the eigenfunctions, eigenvalues of a 1D Derivative operator 
-    with finite and periodic boundary conditions.
+    Class to represent the eigenfunctions, eigenvalues of a 1D Derivative operator
+    on a finite interval with periodic boundary conditions.
 
     Args:
         num_lattice_points: Number of lattice points in the 1D domain
         L: Length of the periodic 1D domain
+        theta: Real number in [0,1] as per the boundary condition (e.g. 0 for periodic, 0.5 for anti-periodic)
     """
 
-    def __init__(self, num_lattice_points, L=1):
+    def __init__(self, num_lattice_points, L=1, theta=0):
         self.num_lattice_points = num_lattice_points
         self.L = L
+        self.a = L / num_lattice_points
+        self.theta = theta
         self.eigenvalues = self._eigenvalues()
 
-
-    def eigenfunction(self, index):
+    def _eigenvalues(self):
         """
-        Private function to return the eigenfunctions of the 1D derivative operator i.e. exp(ikx)
+        Private function to return the eigenvalues of the 1D derivative operator
+        i.e. ik for the k-th eigenfunction exp(ikx) and k = 2*pi*m/L
         """
-        return lambda x: np.exp(self.eigenvalues[index] * x) / np.sqrt(
-            self.num_lattice_points
-        )
+        return 2j * np.pi * (np.fft.fftfreq(self.num_lattice_points, d=self.a) - self.theta / self.L)
 
-    def _eigenvalues(self):
+    def eigenfunction(self, index: np.ndarray):
         """
-        Private function to return the eigenvalues of the 1D derivative operator 
-        i.e. ik for the k-th eigenfunction exp(ikx) and k = 2*pi*m/L
+        Function to return the eigenfunctions of the 1D derivative operator i.e. exp(ikx)
         """
-        return 1j * 2 * np.pi * np.arange(self.num_lattice_points) / self.L
+        index = np.asarray(index)
+
+        if (index < 0).any() or (index >= self.num_lattice_points).any():
```

Comment:
```suggestion
        if (index >= self.num_lattice_points).any():
```
we should allow this to have Python style `[-1]` for last element, etc.